be/src/exec/olap_scanner.cpp

```diff
@@ -152,14 +152,15 @@ Status OlapScanner::_init_tablet_reader_params(
                       ->rowset()
                       ->rowset_meta()
                       ->is_segments_overlapping());
+    
+    bool has_replace_col = false;
+    RETURN_IF_ERROR(_init_return_columns(!single_non_overlapping_version, has_replace_col));
 
-    _tablet_reader_params.direct_mode = single_version || _aggregation;
```

Comment:
Is the following change also can fix the issue?
```
bool unique_key_with_seq =  _tablet->keys_type() == UNIQUE_KEYS && _tablet->tablet_schema().has_sequence_col();
_tablet_reader_params.direct_mode = single_version || (_aggregation && ! unique_key_with_seq) ;
```

Reply:
I  suspect that columns aggregated by replace without SEQ does not work either.

Reply:
> I suspect that columns aggregated by replace without SEQ does not work either.

root cause is _aggregation should not be true in such situation?